File: src/main/java/com/epri/dss/common/impl/CommandLineDSSForms.java
package com.epri.dss.common.impl;

import java.util.List;
import java.util.Scanner;

import javax.swing.JOptionPane;

import com.epri.dss.common.DSSForms;

public class CommandLineDSSForms implements DSSForms {

	private static Scanner sc = new Scanner(System.in);

	private boolean ControlPanelCreated;
//	private static ControlPanel ControlPanel;

	private boolean RebuildHelpForm;

	private CommandLineDSSForms() {

	}

	private static class DSSFormsHolder {
		public static final CommandLineDSSForms INSTANCE = new CommandLineDSSForms();
	}

	public static CommandLineDSSForms getInstance() {
		return DSSFormsHolder.INSTANCE;
	}

	public void createControlPanel() {

	}

	public void exitControlPanel() {

	}

	public void initProgressForm() {

	}

	public void progressCaption(String S) {

	}

	public void progressFormCaption(String S) {

	}

	public void progressHide() {

	}

	public void showControlPanel() {

	}

	public void showHelpForm() {

	}

	public void showAboutBox() {

	}

	public void showPropEditForm() {

	}

	public void showPctProgress(int Count) {

	}

	public void showMessageForm(List<String> S) {

	}

	public int messageDlg(String msg, boolean err) {
		int Result;
		if (err) {
			System.err.println(msg);
			Result = 0;
		} else {
			while (true) {
				String answer;

				System.out.println(msg);
				System.out.print("\nEnter \"Ignore\" or \"Abort\" [Ignore]: ");

				answer = sc.next();
				if (answer.equalsIgnoreCase("abort")) {
					Result = -1;
					break;
				} else if (answer.equalsIgnoreCase("ignore") || answer.length() == 0) {
					Result = 0;
					break;
				}
			}
		}
		return Result;
	}

	public void infoMessageDlg(String msg) {
		System.out.println(msg);
	}

	public String getDSSExeFile() {
		return null;
	}

	public void closeDownForms() {

	}

	public void showTreeView(String Fname) {

	}

	public boolean makeChannelSelection(int NumFieldsToSkip, String Filename) {
		return false;
	}

	public boolean isRebuildHelpForm() {
		return RebuildHelpForm;
	}

	public void setRebuildHelpForm(boolean rebuildHelpForm) {
		RebuildHelpForm = rebuildHelpForm;
	}

}

File: pydss/common/CommandLineDSSForms.py
from dss.common.DSSForms import (DSSForms,)
# from java.util.Scanner import (Scanner,)
# from javax.swing.JOptionPane import (JOptionPane,)
import sys
import sys


class CommandLineDSSForms(object, DSSForms):
    sc = Scanner(sys.stdin)
    controlPanelCreated = None
    # private static ControlPanel ControlPanel;
    rebuildHelpForm = None

    def __init__(self):
        pass

    class DSSFormsHolder(object):
        INSTANCE = CommandLineDSSForms()

    @classmethod
    def getInstance(cls):
        return cls.DSSFormsHolder.INSTANCE

    def createControlPanel(self):
        pass

    def exitControlPanel(self):
        pass

    def initProgressForm(self):
        pass

    def progressCaption(self, s):
        pass

    def progressFormCaption(self, s):
        pass

    def progressHide(self):
        pass

    def showControlPanel(self):
        pass

    def showHelpForm(self):
        pass

    def showAboutBox(self):
        pass

    def showPropEditForm(self):
        pass

    def showPctProgress(self, count):
        pass

    def showMessageForm(self, s):
        pass

    def messageDlg(self, msg, err):
        if err:
            print msg
            Result = 0
        else:
            while True:
                print msg
                sys.stdout.write('\nEnter \"Ignore\" or \"Abort\" [Ignore]: ')
                answer = self.sc.next()
                if answer.equalsIgnoreCase('abort'):
                    Result = -1
                    break
                elif answer.equalsIgnoreCase('ignore') or (len(answer) == 0):
                    Result = 0
                    break
        return Result

    def infoMessageDlg(self, msg):
        print msg

    def getDSSExeFile(self):
        return None

    def closeDownForms(self):
        pass

    def showTreeView(self, fname):
        pass

    def makeChannelSelection(self, numFieldsToSkip, filename):
        return False

    def isRebuildHelpForm(self):
        return self.rebuildHelpForm

    def setRebuildHelpForm(self, rebuild):
        self.rebuildHelpForm = rebuild
